docs/0_concepts.md

```diff
@@ -0,0 +1,79 @@
+---
+title: Serverless Concepts
+description: todo
+layout: Page
+---
+
+# Concepts
+
+Before we begin, let's run through some of the main concepts behind serverless.
+
+- [Serverless Functions](#serverless-functions)
+- [Serverless Events](#serverless-events)
+- [Serverless Resources](#serverless-resources)
+- [Serverless Services](#serverless-services)
+- [Serverless Plugins](#serverless-plugins)
+
+# Serverless Functions
+
+Functions are the unit of deployment in a serverless application
+
+# Serverless Events
+
+Serverless is used to build event driven architecture.
+
+Basically everything which can trigger a function is an event.
+
+Events could be HTTP requests, events fired from a cloud storage (like a S3 bucket), scheduled events, etc.
+
+todo: add links to provider events
+
+# Serverless Resources
+
+Resources are the different pieces that comprise your infrastructure
+
+# Serverless Services
+
+A *Serverless service* is a group of one or multiple functions and any resources they require. By grouping related functions together, it's easier to share code and resources between those functions. Services are also designed to be completely independent, which helps teams develop more quickly without waiting for others.
+
+Each *Serverless service* contains two configuration files:
+
+### [`serverless.yml`](./serverless-yml.md)
+  - Declares a Serverless service
+  - Defines one or multiple functions in the service
+  - Defines the provider the service will be deployed to (and the runtime if provided)
+  - Defines custom plugins to be used
+  - Defines events that trigger each function to execute (e.g. HTTP requests)
+  - Defines one set of resources (e.g. 1 AWS CloudFormation stack) required by the functions in this service
```

Comment:
The resources are merged into the core CloudFormation template. The resources in the `serverless.yml` file are optional.